app/javascript/components/map/map.js

```diff
@@ -37,16 +38,25 @@ const mapStateToProps = (
 
 class MapContainer extends PureComponent {
   componentDidMount() {
-    this.buildMap(this.props);
-    this.props.getLayerSpec();
+    const { mapOptions, getLayerSpec, setMapOptions } = this.props;
+    this.buildMap();
+    getLayerSpec();
+    setMapOptions(mapOptions);
```

Comment:
I am not sure this is needed. If we are getting from the reducer then on mount it will already have them?